### PR TITLE
Start adding support for specifying the initial data in the input file

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

This PR focuses on changes to generic code to support both the current hard-coded ID and to also support reading from the input file. Once the individual systems are converted we will have significant fewer executables and can much more easily add input file tests to CI since they won't increase build costs.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Initialization/SetVariables.hpp [hunk 1]
```diff
@@ -63,13 +67,34 @@ struct SetVariables {
...
+            impl<Metavariables>(make_not_null(&box), *data_or_solution);
+          });
+    } else if constexpr (db::tag_is_retrievable_v<
+                             ::Tags::AnalyticSolutionOrData,
+                             db::DataBox<DbTagsList>>) {
+      impl<Metavariables>(make_not_null(&box),
+                          db::get<::Tags::AnalyticSolutionOrData>(box));
+    } else {
+      ERROR(
+          "Either ::Tags::AnalyticSolutionOrData or BLAH must be in the "
```
Comment (kidder) [inline src/Evolution/Initialization/SetVariables.hpp]: what is BLAH?

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash the one requested change

you can optionally add test coverage for the clone functions

**Comment by nilsdeppe** [comment]
I've rebased, filled in the tag name, and added tests of the clone functions. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
ignoring clang-tidy boost error

**Pull request merged**